### PR TITLE
Enable rebinning for MDHistoWorkspace with original in sliceviewer

### LINKED ISSUE #29086: Workbench Sliceviewer: Don't automatically use original MD workspace on opening sliceviewer
### Expected behavior

Sliceviewer shows the workspace you clicked on.

### Actual behavior

If an MD workspace has an attached workspace it jumps straight to this workspace. If you have non-axis aligned binned the original you don't see the picture you were expecting but the axis-aligned original
 
### Steps to reproduce the behavior
Create an MD workspace:
```python
import math
mdws = CreateMDWorkspace(Dimensions=3, Extents='-10,10,-10,10,-10,10', Names='A,B,C', Units='U,U,U')
FakeMDEventData(InputWorkspace=mdws, PeakParams='100000,-5,-5,0,1')
FakeMDEventData(InputWorkspace=mdws, PeakParams='100000,0,0,0,1')
FakeMDEventData(InputWorkspace=mdws, PeakParams='100000,5,5,0,1')
binned_ws = BinMD(InputWorkspace=mdws, AxisAligned=False, BasisVector0='a,unit,1,1,0',BasisVector1='b,unit,-1,1,0',BasisVector2='c,unit,0,0,1',NormalizeBasisVectors=True,Translation=[-10,-10,0], OutputExtents=[0,math.sqrt(2*20*20),-2,2,-10,10], OutputBins=[100, 100, 1] )
```

Show it in the sliceviewer and the view will be of the original workspace, axis-aligned.

### Platforms affected
All

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

When an MDHistoWorkspace has an attached original workspace then
the rebinning boxes will now be displayed and the binning will take place
from the original workspace. The image will also be dynamically binned as
users zoom.

**To test:**

Use the non-axis-aligned example in the [BinMD](https://docs.mantidproject.org/nightly/algorithms/BinMD-v1.html#usage) docs to produce an MDHistoWorkspace from an MDEventWorkspace. View it in the sliceviewer. It should look like the image on the page with the exception of using a different colormap. As long as the original workspace is present then zooming should dynamically rebin the workspace to the current limits.

Fixes #29086

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/widgets/sliceviewer/peaksviewer/representation/alpha.py [hunk 1]
```diff
@@ -13,13 +13,18 @@
...
     # Apply a linear transform to convert from a distance to an opacity between
     # alpha min & max
-    gradient = (ALPHA_MIN - ALPHA_MAX) / (slicedim_width * VIEW_FRACTION)
-    distance = abs(slicepoint - z)
-    return (gradient * distance) + ALPHA_MAX
+    try:
+        gradient = (ALPHA_MIN - ALPHA_MAX) / (slicedim_width * VIEW_FRACTION)
+        distance = abs(slicepoint - z)
+        return (gradient * distance) + ALPHA_MAX
+    except ArithmeticError:
```
Comment (DavidFair) [inline qt/python/mantidqt/widgets/sliceviewer/peaksviewer/representation/alpha.py:29]: Could we add a unit test to check the div by zero is caught if someone does some overzealous refactoring in the future?

File: qt/python/mantidqt/widgets/sliceviewer/peaksviewer/representation/painter.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
 # constants
 ZOOM_PAD_FRAC = 0.2
+MIN_PAD = 0.05
```
Comment (DavidFair) [inline qt/python/mantidqt/widgets/sliceviewer/peaksviewer/representation/painter.py:17]: Could we comment what these constants do whilst were here?

**Review by DavidFair — CHANGES_REQUESTED** [review]
Some really minor points

I'm going to test this on RHEL7 tomorrow as I suspect it is most likely to throw up any edge cases

File: qt/python/mantidqt/widgets/sliceviewer/peaksviewer/representation/alpha.py (hunk 1 above)
Comment (martyngigg) [reply qt/python/mantidqt/widgets/sliceviewer/peaksviewer/representation/alpha.py:29]: Yes good point!

File: qt/python/mantidqt/widgets/sliceviewer/peaksviewer/representation/painter.py (hunk 2 above)
Comment (martyngigg) [reply qt/python/mantidqt/widgets/sliceviewer/peaksviewer/representation/painter.py:17]: They are constant 😉 
I'll add better comments there :)

**Comment by martyngigg** [comment]
Thanks, @DavidFair. I've addressed those issues now.

**Review by DavidFair — APPROVED** [review]
Thanks for that, lets :shipit:

**Pull request merged**